src/content/React-CSharp/ClientApp/src/components/FetchData.js

```diff
@@ -1,71 +1,65 @@
-import React, { Component } from 'react';
+import React from 'react';
+import { useState, useEffect } from 'react';
 ////#if (IndividualLocalAuth)
 import authService from './api-authorization/AuthorizeService'
 ////#endif
 
-export class FetchData extends Component {
-  static displayName = FetchData.name;
-
-  constructor(props) {
-    super(props);
-    this.state = { forecasts: [], loading: true };
-  }
-
-  componentDidMount() {
-    this.populateWeatherData();
-  }
+export function FetchData() {
+  const [forecasts, setForecasts] = useState([]);
+  const [loading, setLoading] = useState(true);
+  
+  useEffect(async () => {
+    populateWeatherData();
+  }, [])
+  
+  const populateWeatherData = async () => {
+    ////#if (IndividualLocalAuth)
+    const token = await authService.getAccessToken();
+    const response = await fetch('weatherforecast', {
+      headers: !token ? {} : {'Authorization': `Bearer ${token}`}
+    });
+    ////#else
+    const response = await fetch('weatherforecast');
+    ////#endif
+    const data = await response.json();
+    setForecasts(data);
+    setLoading(false);
+  };
 
-  static renderForecastsTable(forecasts) {
+  const renderForecastsTable = (forecasts) => {
     return (
       <table className='table table-striped' aria-labelledby="tableLabel">
         <thead>
-          <tr>
-            <th>Date</th>
-            <th>Temp. (C)</th>
-            <th>Temp. (F)</th>
-            <th>Summary</th>
-          </tr>
+        <tr>
+          <th>Date</th>
+          <th>Temp. (C)</th>
+          <th>Temp. (F)</th>
+          <th>Summary</th>
+        </tr>
         </thead>
         <tbody>
-          {forecasts.map(forecast =>
-            <tr key={forecast.date}>
-              <td>{forecast.date}</td>
-              <td>{forecast.temperatureC}</td>
-              <td>{forecast.temperatureF}</td>
-              <td>{forecast.summary}</td>
-            </tr>
-          )}
+        {forecasts.map(forecast =>
+          <tr key={forecast.date}>
+            <td>{forecast.date}</td>
+            <td>{forecast.temperatureC}</td>
+            <td>{forecast.temperatureF}</td>
+            <td>{forecast.summary}</td>
+          </tr>
+        )}
         </tbody>
       </table>
     );
   }
```

Comment:
I totally agree and in terms of code separation and SRP this is a really good solution. I'm not sure how it can affect the developer experience and the readability of the template's code. So I chose minimalism instead of clean code.

Reply:
sure, sounds reasonable